src/dotnet/OpenAPI.Validator/JsonConverters/PathItemRefConverter.cs

```diff
@@ -26,13 +26,13 @@ public override bool CanConvert(System.Type objectType)
             JsonSerializer serializer)
         {
             // is the leaf an vendor extension? "x-*..."
-            if (reader == null || reader.Path.Substring(reader.Path.LastIndexOf(".", StringComparison.Ordinal) + 1).StartsWith("x-", StringComparison.CurrentCulture))
+            JObject jobject = JObject.Load(reader);
```

Comment:
You can add some unit tests to cover the new feature.

Reply:
You can add some comments here to list which case is invalid.

Reply:
Why you move `JObject jobject ...`this line from bottom to here.

Reply:
That's the line causing the problem. it return too soon that the parse process doesn't move on, causing it runs the same step next time. and  causing next step has null pointer error.